src/main/java/emu/grasscutter/game/quest/exec/ExecRefreshGroupSuite.java

```diff
@@ -16,32 +13,18 @@ public class ExecRefreshGroupSuite extends QuestExecHandler {
     public boolean execute(GameQuest quest, QuestData.QuestExecParam condition, String... paramStr) {
         var sceneId = Integer.parseInt(paramStr[0]);
         var entries = paramStr[1].split(";");
+
+        boolean result = true;
         for(var entry : entries) {
             var groupId = Integer.parseInt(entry.split(",")[0]);
             var suiteId = Integer.parseInt(entry.split(",")[1]);
 
-            var scriptManager = quest.getOwner().getScene().getScriptManager();
-
-            // refresh immediately if player is in this scene
-            if (quest.getOwner().getScene().getId() == sceneId) {
-                var targetGroupInstance = scriptManager.getGroupInstanceById(groupId);
-                if (targetGroupInstance == null) {
-                    Grasscutter.getLogger().warn("trying to load unknown group {} in scene {}", groupId, sceneId);
-                } else {
-                    suiteId = scriptManager.refreshGroup(targetGroupInstance, suiteId, false); //If suiteId is zero, the value of suiteId changes
-                    quest.getOwner().sendPacket(new PacketGroupSuiteNotify(groupId, suiteId));
-                }
+            if(!quest.getOwner().getWorld().getSceneById(sceneId).getScriptManager().refreshGroupSuite(groupId, suiteId, quest)) {
```

Comment:
I think you can remove the if and directly do return refreshGroupSuite()
Unless you change it before it, I will probably change it at the merge.

Reply:
Not really, it's a for loop, i can't return directly, the quest can specify more than on group to refresh

Reply:
Oh yeah, didn't notice that, between all the changes.

Not part of your changes, but it might be good to store the result of entry.split(",") In a val, to half the string operations xD

Reply:
Done xD